src/math/web_mercator.rs

```diff
@@ -0,0 +1,143 @@
+//! Calculations with Web Mercator coordinates and tiles.
+
+use nalgebra::Vector2;
+use nav_types::WGS84;
+use std::f64::consts::{FRAC_1_PI, PI};
+
+/// 2.0 * E.powf(PI).arctan() - FRAC_PI_2;
+/// In degrees, it's 85.051129 (cf. Wikipedia)
+const LAT_BOUND_RAD: f64 = 1.484_422_229_745_332_4;
+
+/// LAT_BOUND_SIN = sin(LAT_BOUND_RAD)
+const LAT_BOUND_SIN: f64 = 0.996_272_076_220_75;
+
+const TWO_PI: f64 = 2.0 * PI;
+const FOUR_PI: f64 = 4.0 * PI;
+const FOUR_PI_INV: f64 = 0.25 * FRAC_1_PI;
+const TILE_SIZE: u32 = 256;
+
+/// The max zoom level is currently 23 because of an implementation choice,
+/// namely fitting TILE_SIZE << MAX_ZOOM in an u32, but theoretically nothing
+/// stops us from going deeper.
+pub const MAX_ZOOM: u8 = 23;
+
+/// A Web Mercator coordinate. Essentially a position in a 2D map of the world.
+#[derive(Copy, Clone, Debug, PartialEq, PartialOrd)]
+pub struct WebMercatorCoord {
+    /// Implementation detail: This is normalized to [0, 1), so not zoom level 0.
+    /// This makes calculations a bit simpler.
+    xy: Vector2<f64>,
+}
```

Comment:
What do you think about just doing
```rust
pub struct WebMercatorCoord(Vector2<f64>);
```
or
```rust
pub struct WebMercatorCoord(pub Vector2<f64>);
```
depending on whether you want to make the inner vector public or not?

Reply:
I don't have a preference worth mentioning here, but I like giving it a name as a means of additional documentation (I changed it to something more meaningful). If there's a reason a tuple struct would be a better fit I'm happy to change it though.